bazel/astore/defs.bzl

```diff
@@ -60,3 +60,47 @@ astore_upload = rule(
 With this rule, you can easily upload the output of a build rule
 to an artifact store.""",
 )
+
+def _astore_download(ctx):
+    push = ctx.actions.declare_file("astore_download.sh")
+    inputs = [ctx.executable._astore_client]
+    template = ctx.file._astore_download_file
+    ctx.actions.expand_template(
+        template = template,
+        output = push,
+        substitutions = {
+            "{astore}": ctx.executable._astore_client.short_path,
+            "{file}": ctx.attr.file,
+            "{destination}": ctx.attr.destination
+        },
+        is_executable = True,
+    )
+    return [DefaultInfo(executable = push, runfiles = ctx.runfiles(inputs))]
+
+astore_download = rule(
+    implementation = _astore_download,
+    attrs = {
+        "destination": attr.string(
+            doc = "The file will be downloaded from astore to this path.",
```

Comment:
is this mandatory? the -o parameter is always provided in the script

Either:
* add `mandatory = True` or...
* make the command tolerant to not having a destination specified.